src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -244,17 +253,26 @@ public void closeAll() {
     closeStreamingClient();
   }
 
+  /**
+   * This function is called during rebalance. Please ensure we dont wipe of state when we are
+   * invoking this function.
+   *
+   * <p>The channel is not closed since we would need to re-initiate the channel.
+   *
+   * <p>Also, wiping off partitionsToChannel map would mean we would need to reinitialise and we
+   * will lose processedOffset information.
+   *
+   * <p>We will only close the client.
```

Comment:
I don't think this will work, closing the client will close all channels

Reply:
same as above.

Reply:
no, what I mean here is that all the channels will be closed as well when you close the client, so what you said here and at line 260 is not correct

Reply:
Also, can we update `closeAll` to call `closeStreamingClient` directly instead of closing channels one by one?

Reply:
Hmm okay, I think what you are saying is we dont need to close client but we can close all channels. so that we dont need to reopen the client?

Agreed that line 260 is misleading..

Reply:
Let me clean this up..